wrangler-service/src/main/java/io/cdap/wrangler/service/directive/AbstractDirectiveHandler.java

```diff
@@ -154,10 +157,19 @@ protected <E extends Exception> List<Row> executeDirectives(
   protected DirectiveExecutionResponse generateExecutionResponse(
     List<Row> rows, int limit) throws Exception {
     List<Map<String, Object>> values = new ArrayList<>(rows.size());
-    Map<String, String> types = new HashMap<>();
-    Set<String> headers = new LinkedHashSet<>();
-    SchemaConverter convertor = new SchemaConverter();
-
+    Map<String, String> types = new LinkedHashMap<>();
+
+    Schema outputSchema = TRANSIENT_STORE.get(TransientStoreKeys.OUTPUT_SCHEMA) != null ?
+      TRANSIENT_STORE.get(TransientStoreKeys.OUTPUT_SCHEMA) : TRANSIENT_STORE.get(TransientStoreKeys.INPUT_SCHEMA);
```

Comment:
In which cases OUTPUT_SCHEMA will be null? This "else" part seems problematic for me.

Reply:
Initially only input schema is set in transient store (when workspace created for ex)